From 1ca32188721296c8ac26cf256d04cc15756c6e3b Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Fri, 9 Jun 2023 15:47:16 +0000
Subject: [PATCH] [static-stdlib] Add static executable support for WASI target

This patch adds static-executble-args.lnk file for WASI target
---
 .../public/Resources/wasi/static-executable-args.lnk  | 11 +++++++++++
 stdlib/public/runtime/CMakeLists.txt                  |  2 +-
 2 files changed, 12 insertions(+), 1 deletion(-)
 create mode 100644 stdlib/public/Resources/wasi/static-executable-args.lnk

diff --git a/stdlib/public/Resources/wasi/static-executable-args.lnk b/stdlib/public/Resources/wasi/static-executable-args.lnk
new file mode 100644
index 00000000000..1500f94f891
--- /dev/null
+++ b/stdlib/public/Resources/wasi/static-executable-args.lnk
@@ -0,0 +1,11 @@
+-static
+-lswiftSwiftOnoneSupport
+-ldl
+-lstdc++
+-lm
+-lwasi-emulated-mman
+-lwasi-emulated-signal
+-lwasi-emulated-process-clocks
+-Xlinker --error-limit=0
+-Xlinker --no-gc-sections
+-Xlinker --threads=1
diff --git a/stdlib/public/runtime/CMakeLists.txt b/stdlib/public/runtime/CMakeLists.txt
index 4a9410bcd1e..d7c352deb32 100644
--- a/stdlib/public/runtime/CMakeLists.txt
+++ b/stdlib/public/runtime/CMakeLists.txt
@@ -106,7 +106,7 @@ if(SWIFT_BUILD_STATIC_STDLIB)
   set(static_binary_lnk_file_list)
 
   foreach(sdk ${SWIFT_SDKS})
-    if(NOT "${sdk}" STREQUAL "LINUX")
+    if(NOT "${sdk}" STREQUAL "LINUX" AND NOT "${sdk}" STREQUAL "WASI")
       continue()
     endif()
 
-- 
2.40.0

